src/sources/dnstap/dns_message_parser.rs

```diff
@@ -0,0 +1,2055 @@
+                            for keyvalue in vec_keyvalue.iter() {
+                                final_issuer.push_str("; ");
+                                final_issuer.push_str(&keyvalue.key());
+                                final_issuer.push('=');
+                                final_issuer.push_str(&keyvalue.value());
+                            }
+                        }
+                        final_issuer.trim_end().to_string()
+                    }
+                    Value::Unknown(unknown) => std::str::from_utf8(unknown)
+                        .context(Utf8ParsingError)?
+                        .to_string(),
+                }
+            );
+            Ok((Some(caa_rdata), None))
+        }
+
+        RData::TLSA(tlsa) => {
+            let tlsa_rdata = format!(
+                "{} {} {} {}",
+                u8::from(tlsa.cert_usage()),
+                u8::from(tlsa.selector()),
+                u8::from(tlsa.matching()),
+                HEXUPPER.encode(&tlsa.cert_data())
+            );
+            Ok((Some(tlsa_rdata), None))
+        }
+        RData::SSHFP(sshfp) => {
+            let sshfp_rdata = format!(
+                "{} {} {}",
+                Into::<u8>::into(sshfp.algorithm()),
+                Into::<u8>::into(sshfp.fingerprint_type()),
+                HEXUPPER.encode(&sshfp.fingerprint())
+            );
+            Ok((Some(sshfp_rdata), None))
+        }
+        RData::NAPTR(naptr) => {
+            let naptr_rdata = format!(
+                r#"{} {} "{}" "{}" "{}" {}"#, // why the " escaping desnt work?
+                naptr.order(),
+                naptr.preference(),
+                escape_string_for_text_representation(
+                    std::str::from_utf8(naptr.flags())
+                        .context(Utf8ParsingError)?
+                        .to_string()
+                ),
+                escape_string_for_text_representation(
+                    std::str::from_utf8(naptr.services())
+                        .context(Utf8ParsingError)?
+                        .to_string()
+                ),
+                escape_string_for_text_representation(
+                    std::str::from_utf8(naptr.regexp())
+                        .context(Utf8ParsingError)?
+                        .to_string()
+                ),
+                naptr.replacement().to_utf8()
+            );
+            Ok((Some(naptr_rdata), None))
+        }
+        RData::DNSSEC(dnssec) => match dnssec {
+            DNSSECRData::DS(ds) => {
+                let ds_rdata = format!(
+                    "{} {} {} {}",
+                    ds.key_tag(),
+                    u8::from(ds.algorithm()),
+                    u8::from(ds.digest_type()),
+                    HEXUPPER.encode(ds.digest())
+                );
+                Ok((Some(ds_rdata), None))
+            }
+            DNSSECRData::DNSKEY(dnskey) => {
+                let dnskey_rdata = format!(
+                    "{} 3 {} {}",
```

Comment:
`3` is is a fixed Protocol value for `DNSKEY` rdata. See [DNSKEY RDATA Wire Format](https://tools.ietf.org/html/rfc4034#section-2.1) for details please.

Reply:
👍 could we add a comment and maybe a link to that here? I realize it might be obvious to people already familiar with dnstap and dnssec but it'll help maintainers (including myself 😄 ) that are less familiar.

Reply:
Certainly 👍

Reply:
Updated.